`test_external_inspection` fails on Ubuntu

# Bug report

### Bug description:

Example: https://github.com/python/cpython/actions/runs/14827808058/job/41623568354?pr=133365

```
0:12:02 load avg: 6.85 [487/491/3] test_external_inspection failed (3 failures) -- running (2): test.test_concurrent_futures.test_process_pool (1 min 53 sec), test.test_multiprocessing_spawn.test_manager (1 min 10 sec)
test_async_gather_remote_stack_trace (test.test_external_inspection.TestGetStackTrace.test_async_gather_remote_stack_trace) ... ok
test_async_global_awaited_by (test.test_external_inspection.TestGetStackTrace.test_async_global_awaited_by) ... ok
test_async_remote_stack_trace (test.test_external_inspection.TestGetStackTrace.test_async_remote_stack_trace) ... 
  test_async_remote_stack_trace (test.test_external_inspection.TestGetStackTrace.test_async_remote_stack_trace) (task_factory_variant='asyncio.new_event_loop') ... FAIL
  test_async_remote_stack_trace (test.test_external_inspection.TestGetStackTrace.test_async_remote_stack_trace) (task_factory_variant='new_eager_loop') ... FAIL
test_async_staggered_race_remote_stack_trace (test.test_external_inspection.TestGetStackTrace.test_async_staggered_race_remote_stack_trace) ... ok
test_asyncgen_remote_stack_trace (test.test_external_inspection.TestGetStackTrace.test_asyncgen_remote_stack_trace) ... ok
test_remote_stack_trace (test.test_external_inspection.TestGetStackTrace.test_remote_stack_trace) ... FAIL
test_self_trace (test.test_external_inspection.TestGetStackTrace.test_self_trace) ... ok

======================================================================
FAIL: test_async_remote_stack_trace (test.test_external_inspection.TestGetStackTrace.test_async_remote_stack_trace) (task_factory_variant='asyncio.new_event_loop')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/cpython/cpython-ro-srcdir/Lib/test/test_external_inspection.py", line 270, in test_async_remote_stack_trace
    self.assertEqual(stack_trace, expected_stack_trace)
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: Lists differ: [[('c[62 chars]y', 10), ('c4', '/tmp/test_python_ckrn9tbc/tmp[1269 chars]]]]]] != [[('c[62 chars]y', 11), ('c4', '/tmp/test_python_ckrn9tbc/tmp[1284 chars]]]]]]

First differing element 0:
[('c5[61 chars]y', 10), ('c4', '/tmp/test_python_ckrn9tbc/tmp[179 chars] 21)]
[('c5[61 chars]y', 11), ('c4', '/tmp/test_python_ckrn9tbc/tmp[179 chars] 21)]

Diff is 1984 characters long. Set self.maxDiff to None to see it.

======================================================================
FAIL: test_async_remote_stack_trace (test.test_external_inspection.TestGetStackTrace.test_async_remote_stack_trace) (task_factory_variant='new_eager_loop')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/cpython/cpython-ro-srcdir/Lib/test/test_external_inspection.py", line 270, in test_async_remote_stack_trace
    self.assertEqual(stack_trace, expected_stack_trace)
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: Lists differ: [[('c[62 chars]y', 10), ('c4', '/tmp/test_python_ckrn9tbc/tmp[1269 chars]]]]]] != [[('c[62 chars]y', 11), ('c4', '/tmp/test_python_ckrn9tbc/tmp[1284 chars]]]]]]

First differing element 0:
[('c5[61 chars]y', 10), ('c4', '/tmp/test_python_ckrn9tbc/tmp[179 chars] 21)]
[('c5[61 chars]y', 11), ('c4', '/tmp/test_python_ckrn9tbc/tmp[179 chars] 21)]

Diff is 1984 characters long. Set self.maxDiff to None to see it.

======================================================================
FAIL: test_remote_stack_trace (test.test_external_inspection.TestGetStackTrace.test_remote_stack_trace)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/cpython/cpython-ro-srcdir/Lib/test/test_external_inspection.py", line 104, in test_remote_stack_trace
    self.assertEqual(stack_trace, expected_stack_trace)
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: Lists differ: [('fo[62 chars]y', 14), ('baz', '/tmp/test_python_ckrn9tbc/tm[186 chars] 17)] != [('fo[62 chars]y', 15), ('baz', '/tmp/test_python_ckrn9tbc/tm[186 chars] 17)]

First differing element 0:
('foo', '/tmp/test_python_ckrn9tbc/tmpf0v2rgso/script_pkg/script.py', 14)
('foo', '/tmp/test_python_ckrn9tbc/tmpf0v2rgso/script_pkg/script.py', 15)

- [('foo', '/tmp/test_python_ckrn9tbc/tmpf0v2rgso/script_pkg/script.py', 14),
?                                                                         ^

+ [('foo', '/tmp/test_python_ckrn9tbc/tmpf0v2rgso/script_pkg/script.py', 15),
?                                                                         ^

   ('baz', '/tmp/test_python_ckrn9tbc/tmpf0v2rgso/script_pkg/script.py', 11),
   ('bar', '/tmp/test_python_ckrn9tbc/tmpf0v2rgso/script_pkg/script.py', 9),
   ('<module>', '/tmp/test_python_ckrn9tbc/tmpf0v2rgso/script_pkg/script.py', 17)]

----------------------------------------------------------------------
Ran 7 tests in 6.071s

FAILED (failures=3)
test test_external_inspection failed
```

Came across this failure in my PR to add zstd Python code.

### CPython versions tested on:

CPython main branch

### Operating systems tested on:

Linux

<!-- gh-linked-prs -->
### Linked PRs
* gh-133422
* gh-133433
<!-- /gh-linked-prs -->

--- a/Lib/test/test_external_inspection.py
+++ b/Lib/test/test_external_inspection.py
@@ -59,8 +59,7 @@ class TestGetStackTrace(unittest.TestCase):
                 foo()
 
             def foo():
-                sock.sendall(b"ready")
-                time.sleep(1000)
+                sock.sendall(b"ready"); time.sleep(1000)  # same line number
 
             bar()
             """
@@ -96,10 +95,10 @@ class TestGetStackTrace(unittest.TestCase):
                 p.wait(timeout=SHORT_TIMEOUT)
 
             expected_stack_trace = [
-                ("foo", script_name, 15),
+                ("foo", script_name, 14),
                 ("baz", script_name, 11),
                 ("bar", script_name, 9),
-                ("<module>", script_name, 17),
+                ("<module>", script_name, 16),
             ]
             self.assertEqual(stack_trace, expected_stack_trace)
 

